commands/utils/packagehandlers/common.go

```diff
@@ -0,0 +1,63 @@
+package packagehandlers
+
+import (
+	"fmt"
+	"github.com/jfrog/frogbot/commands/utils"
+	"github.com/jfrog/jfrog-cli-core/v2/utils/coreutils"
+	"github.com/jfrog/jfrog-client-go/utils/log"
+	"os/exec"
+	"strings"
+)
+
+// PackageHandler interface to hold operations on packages
+type PackageHandler interface {
+	UpdateDependency(details *utils.FixDetails) (bool, error)
+	updateDirectDependency(fixDetails *utils.FixDetails, extraArgs ...string) (fixSupported bool, err error)
```

Comment:
Do we really need to seperate these functionalities? Personally I believe we should use `UpdateDependency` only.

Reply:
UpdateDependency is the only public function,
and then we have 2 private functions for direct & indirect as most of the time will require different logic.